homeassistant/components/eight_sleep/__init__.py

```diff
@@ -170,9 +199,25 @@ def __init__(
 
         mapped_name = NAME_MAP.get(sensor, sensor.replace("_", " ").title())
         if self._user_obj is not None:
-            mapped_name = f"{self._user_obj.side.title()} {mapped_name}"
-
-        self._attr_name = f"Eight {mapped_name}"
-        self._attr_unique_id = (
-            f"{_get_device_unique_id(eight, self._user_obj)}.{sensor}"
-        )
+            assert self._user_obj.user_profile
+            name = f"{self._user_obj.user_profile['firstName']}'s {mapped_name}"
+            self._attr_name = name
+        else:
+            self._attr_name = f"Eight Sleep {mapped_name}"
+        unique_id = f"{_get_device_unique_id(eight, self._user_obj)}.{sensor}"
+        self._attr_unique_id = unique_id
+        identifiers = {(DOMAIN, _get_device_unique_id(eight, self._user_obj))}
```

Comment:
As above.

Reply:
I split it into two lines because otherwise it would be three and I thought it was easier to read. Do you still think it's better to combine?